catroid/src/androidTest/java/org/catrobat/catroid/test/content/actions/SetLookActionTest.java

```diff
@@ -100,4 +101,33 @@ public void testSetLook() {
 		action.act(1.0f);
 		assertEquals("Action didn't set the LookData", lookData, sprite.look.getLookData());
 	}
+
+	public void testSetLookWithFormula() {
+		Sprite sprite = new SingleSprite("new sprite");
+		project.getDefaultScene().addSprite(sprite);
+
+		LookData firstLookData = new LookData();
+		LookData secondLookData = new LookData();
+		firstLookData.setLookFilename(testImage.getName());
+		secondLookData.setLookFilename(testImage.getName());
+		firstLookData.setLookName("firstTestLook");
+		secondLookData.setLookName("secondTestLook");
+
+		sprite.getLookDataList().add(firstLookData);
+		sprite.getLookDataList().add(secondLookData);
+
+		Formula numberFormula = new Formula(2);
+
+		ActionFactory factory = sprite.getActionFactory();
+		Action action = factory.createSetLookAction(sprite, firstLookData, false, numberFormula);
+		action.act(1.0f);
+		assertEquals("Action didn't set the LookData correctly from the number formula", secondLookData, sprite.look
+				.getLookData());
+
+		Formula stringFormula = new Formula("firstTestLook");
```

Comment:
According to the article series on testable code this part should probably be put in a seperate test, as there should not be acting in between asserts (anatomy of a testcase: assemble - act - assert).